### PR TITLE
feat: add process definition search controller and service infrastucture 

### PR DESCRIPTION
**Pull request by ralfpuchert**
## Description

* Add Process definition QueryController and test
* Add Process definition infrastructure for query, filter and sort
* Add Process definition service infrastructure
* Implementation of filter, sort and CamundaClient follows in later PR's in will also be merged into `operate/c8-process-definition-search-20652` feature branch
* Add OpenAPI description in `rest-api.yaml` (@pepopowitz , @camunda/tech-writers can you review this, please)

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -637,7 +637,47 @@ paths:
             application/problem+json:
               schema:
                 $ref: "#/components/schemas/ProblemDetail"
-
+  /process-definitions/search:
+    post:
+      tags:
+        - Process definition
+      summary: process definitions (alpha)
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:644]:
```suggestion
      summary: Search process definitions (alpha)
```

:x: Summary should be `Title case`
🔧 This feels like a more complete summary of the endpoint.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -637,7 +637,47 @@ paths:
...
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/ProcessDefinitionSearchQueryRequest"
+      responses:
+        "200":
+          description: >
+            The Process Definition Search successful response.
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:661]:
```suggestion
            The process definition search successful response.
```

:x: Remove capitalization

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -637,7 +637,47 @@ paths:
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ProcessDefinitionSearchQueryResponse"
+        "400":
+          description: >
+            The Process Definition Search Query failed.
```
Comment (pepopowitz) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:668]:
```suggestion
            The process definition search query failed.
```

:x: Remove capitalization

**Review by pepopowitz — APPROVED** [review]
👍 A few small changes to conform with [our spec standards](https://github.com/camunda/camunda/blob/main/docs/rest-controller.md#openapi-extension), but these descriptions are great, thank you!

**Review by ghost — APPROVED** [review]

**Pull request merged**